src/System.Data.Common/src/System/Data/Common/DbConnection.cs

```diff
@@ -103,6 +103,15 @@ protected virtual void OnStateChange(StateChangeEventArgs stateChange)
             StateChange?.Invoke(this, stateChange);
         }
 
+        protected bool StateChangeHasSubscribers
+        {
+            get
+            {
+                var stateChange = StateChange;
+                return stateChange != null && stateChange.GetInvocationList().Length > 0;
```

Comment:
Thanks, wasn't aware of that... I was unable to find a definite answer on whether simply checking for null is enough, or whether there's a situation where `GetInvocationList()` returns an empty array.